readme.md

```diff
@@ -50,14 +50,14 @@ partial(2); //=> 42
 
 ## Currying
 
-> The process of converting a function with multiple arity into the same function with an arity of one. Not to be confused with partial application, which can produce a function with an arity greater than one.
+> The process of converting a function of multiple parameters into one, which, when called with fewer parameters, returns a new function that awaits the remaining parameters.  It's important to note that this returned function, if there are still more than one parameter unspecified, is also curried.  Strictly speaking, it is only currying when the returned function accepts just one parameter, but many implementations allow for multiple parameters at each level.  This is not to be confused with partial application, in which the returned function still requires all remaining parameters to be supplied.
```

Comment:
FTR, the wordy definition was mine.  I don't know the overall plan for this document, but I might suggest that the best bet would be to have a brief definition that captures the essence and then if necessary a few additional sentences which clarify it, add caveats, whatever.  (And of course an example or two to follow that.)  With that said, I would nominate the definition from @davidchambers as the key definition, but would back it up, perhaps with something like this:

> _Currying is the transformation from a function with multiple arguments to a function which takes its arguments one at a time. If `f` is a ternary function and `g` is the result of currying `f`, `g(a)(b)(c)` is equivalent to `f(a, b, c)`._
> 
> Some versions of currying allow for additional ways to call, so that these are all equivalent: `g(a, b, c)`, `g(a, b)(c)`, `g(a)(b, c)`, `g(a)(b)(c)`.  Currying is not to be confused with partial application:  If you partially apply the first parameter to `f`, you get a plain binary function.  If you curry `f` and apply the first parameter, you get a new _curried_ function.  That is, `g(a)` is also a curried function. 
> 
> ```js
> let sum = (a, b, c) =>  a + b + c;
> let curriedSum = a => b => c => a + b + c;
> curriedSum(15)(25)(2) //=> 42
> ```